compiler/rustc_mir/src/borrow_check/region_infer/mod.rs

```diff
@@ -1225,7 +1225,9 @@ impl<'tcx> RegionInferenceContext<'tcx> {
     /// it. However, it works pretty well in practice. In particular,
     /// this is needed to deal with projection outlives bounds like
     ///
-    ///     <T as Foo<'0>>::Item: '1
+    /// ```ignore (MIR syntax)
```

Comment:
I'm not 100% sure this is MIR; is that indeed what's being shown here?

Reply:
this is not mir, this `'0` is an compiler generated region variable, so you can think of it as `<T as Foo<'a>>::Item: 'b`
where `'a` and `'b` are created by the compiler.